src/js/calculate.js

```diff
@@ -222,7 +265,7 @@ export default {
                 this.form.validateInput( control );
             }
         } else {
-            // This is okay for a setvalue/xforms-value-changed directive (may be no form control)
+            // This is okay for an xforms-value-changed action (may be no form control)
```

Comment:
Just in case this is a useful comment in the future, perhaps move to the if(!control) block.

This `else{}` can then be removed, I think